src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelector.java

```diff
@@ -43,7 +44,27 @@
  */
 class GroovyCallSiteSelector {
 
-    private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Object[] parameters) {
+    private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Object[] parameters, boolean varargs) {
+        if (varargs) {
+            int fixedLen = parameterTypes.length - 1;
+            Class<?> componentType = parameterTypes[fixedLen].getComponentType();
+            assert componentType != null;
+            int arrayLength = parameters.length - fixedLen;
+            if (arrayLength >= 0) {
+                if (arrayLength == 1 && parameterTypes[fixedLen].isInstance(parameters[fixedLen])) {
+                    // not a varargs call
```

Comment:
🐛  Rather than redefining an argument in a fairly complex code block, let's move this to a separate method returning the new parameters (hard to follow otherwise).